smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/httpResponse/bindingTraits/HTTPResponseTraitWithHTTPPayload.kt

```diff
@@ -82,9 +86,7 @@ class XMLHttpResponseTraitWithHttpPayload(
             ShapeType.STRUCTURE, ShapeType.UNION -> {
                 if (target.hasTrait<StreamingTrait>()) {
                     writer.openBlock("if case .stream(let stream) = httpResponse.body {", "}") {
-                        writer.declareSection(HttpResponseTraitWithHttpPayload.MessageDecoderSectionId) {
-                            writer.write("let messageDecoder: \$D", ClientRuntimeTypes.EventStream.MessageDecoder)
-                        }
+                        writer.write("let messageDecoder = \$N()", customizations.messageDecoderSymbol)
```

Comment:
The message decoder type is read from the customizations, instead of using a section writer which is super confusing and carries a lot of extra weight elsewhere in the code.